Bio/Blast/NCBIWWW.py

```diff
@@ -276,26 +278,72 @@ def qblast(
 
         request = Request(url_base, message, {"User-Agent": "BiopythonClient"})
         handle = urlopen(request)
-        results = handle.read().decode()
+        results = handle.readlines()
 
         # Can see an "\n\n" page while results are in progress,
         # if so just wait a bit longer...
-        if results == "\n\n":
+        if results == [b"\n\n"]:
             continue
-        # XML results don't have the Status tag when finished
-        if "Status=" not in results:
-            break
-        i = results.index("Status=")
-        j = results.index("\n", i)
-        status = results[i + len("Status=") : j].strip()
-        if status.upper() == "READY":
+
+        # Iterate over the results to know if the request is complete.
+        is_done = False
+        for line_no, line in enumerate(results):
+            # Has a Status tag and Status is READY
+            if b"Status=" in line:
+                status = results[line_no + 1].strip()
+                if status.upper() == "READY":
+                    is_done = True
+                    break
+        # Or, XML results don't have the Status tag when finished
+        else:
+            is_done = True
+
+        if is_done:
             break
-    return StringIO(results)
+    return _sanitize_qblast_xml(results)
```

Comment:
`return _sanitize_qblast_xml(results) if format_type == 'XML' else StringIO(results)` ?
Without having looked into the code closely, but couldn't the result be something else than XML?

Reply:
Yes, ``format_type "HTML", "Text", "ASN.1", or "XML". Def. "XML".``

So the default is XML (ought to be binary but parsers generally accept text mode), but there is also a text mode etc.

Probably ASN.1 ought to be binary mode too, but in my experience rarely used - so that could currently be broken.

Reply:
Very good point, I completely overlooked that.